aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1009,7 +1009,12 @@ static void pre_check_gradient(const Tensor& self, c10::optional<int64_t> spacin
   // Helper for gradient function to make sure input data satisfies prerequisites
   TORCH_CHECK(self.scalar_type() != ScalarType::Byte, "torch.gradient does not support uint8 input.");
   if (spacing_size.has_value() && !dim.has_value()) {
-    TORCH_CHECK(spacing_size.value() == 1 || spacing_size.value() == self.dim(), "torch.gradient expected spacing to be unspecified, a scalar or a list of length ", self.dim(), " but got a list of length ", spacing_size.value());
+    // NOTE: If spacing was given as a scalar, the callers of this function
+    // create a spacing vector of the expected size, and this check passes
```

Comment:
"spacing vector" here is "dim" right?

Reply:
No I was referring to the "spacing" arg. Here's an example of one of the callers creating a vector when spacing (named `unit_size` in this impl) is given as a scalar:

https://github.com/pytorch/pytorch/blob/639060cb0b719a652edd74981fe98b927a314cab/aten/src/ATen/native/ReduceOps.cpp#L1144-L1148

Reply:
Thanks for the explanation!